pkg/controller/migmigration/storage.go

```diff
@@ -354,8 +364,33 @@ func (t *Task) swapJobsPVCRefs(client k8sclient.Client, mapping pvcNameMapping)
 			}
 			continue
 		}
-		for _, job := range list.Items {
+		for i := range list.Items {
+			oldJob := &list.Items[i]
+			// if job is already complete, skip
+			if isJobComplete(oldJob) {
+				continue
+			}
 			isFailed := false
+			job := &batchv1.Job{}
+			job.Namespace = oldJob.Namespace
+			job.ObjectMeta.Labels = oldJob.Labels
+			job.ObjectMeta.Annotations = oldJob.Annotations
+			delete(job.Labels, "job-name")
+			delete(job.Labels, "controller-uid")
+			if oldJob.Spec.DeepCopy() != nil {
+				job.Spec = *oldJob.Spec.DeepCopy()
```

Comment:
This is performing the DeepCopy() operation twice if its output is not nil. The first call should save the output into a local variable and use that result in the assignment here.